### PR TITLE
change to_check.guild to getattr()

### LINKED ISSUE #3100: is_automod_immune errors when User object passed
#### What were you trying to do?

Use the automod_immune method

#### What were you expecting to happen?

Return a boolean

#### What actually happened?

Errors

#### How can we reproduce this issue?

Pass a User object into the method

### PR DESCRIPTION
**Pull request by Bakersbakebread**
### Type

- [x] Bugfix
- [ ] Enhancement
- [ ] New feature

### Description of the changes
Replaces to_check.guild with a getattr() 

Edit: fixes #3100 
~Kowlin

### CODE REVIEW

**Review by mikeshardmind — CHANGES_REQUESTED** [review]
As discussed in discord, this should actually return `True` in DMs (and docstring needs updating for this too)

Edit: hold on this, more discussion in discord

**Comment by Kowlin** [comment]
Fixes #3100

File: changelog.d/3100.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+change is_autmod_immune handling of guild check and support for checking webhooks
```
Comment (mikeshardmind) [inline changelog.d/3100.bugfix.rst]:
```suggestion
fix ``is_automod_immune`` handling of guild check and support for checking webhooks
```

**Review by mikeshardmind — APPROVED** [review]
Thanks for the quick follow up + discussion on discord.

**Pull request merged**